app/helpers/address_helper.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module AddressHelper
+  def postcode_lookup(postcode)
+    postcode_url = "#{api_url}&postcode=#{postcode}&key=#{ENV['ORDNANCE_SURVEY_PLACES_API_KEY']}"
+
+    response = Typhoeus.get(postcode_url)
+
+    results = []
+    if response.code == 200
+      addresses = JSON.parse(response.body)["results"]
+      addresses.each do |address|
+        results << { address.dig("LPI", "UPRN") => address.dig("LPI", "ADDRESS") }
+      end
+    else
+      results << { response.code => I18n.t("address_api.error_#{response.code}") }
+    end
```

Comment:
Currently this method returns `[{ 500 => "Error 500" }]` or `[{ "uprn_1" => "10 Downing St..." }, ...]`. When we use this, we must check if any of the keys are non-200 status codes, which seems like unnecessary indirection.

I wouldn't be afraid to create new classes if it helps here. You might find this interface easier to express if you used a class instead of a module, since it could have state, and you could write methods that use that state e.g. `address_response.ok?` and `address_response.addresses`.

For example

```ruby
class AddressLookup
    def candidate_addresses
        @candidate_addresses ||= begin
        postcode_url = "#{API_URL}&postcode=#{postcode}&key=#{Rails.application.secrets.ordinance_survey_api_key}"
        response = Typhoeus.get(postcode_url)
        raise AddressLookupError unless response.status == 200
        present_addresses response
        end
    end

private

    API_URL = "".freeze

    def present_addresses(response)
        JSON.parse(response.body).fetch("results", []).map { ... }
    end
end
```